compaction/task_manager_module.cc

```diff
@@ -300,7 +303,8 @@ future<> table_major_keyspace_compaction_task_impl::run() {
     co_await wait_for_your_turn(_cv, _current_task, _status.id);
     tasks::task_info info{_status.id, _status.shard};
     co_await run_on_table("force_keyspace_compaction", _db, _status.keyspace, _ti, [info] (replica::table& t) {
-        return t.compact_all_sstables(info);
+        // All tables were already flushed in shard_major_keyspace_compaction_task_impl
+        return t.compact_all_sstables(info, replica::table::do_flush::no);
```

Comment:
What does "table_major_keyspace_compaction_task_impl" mean?

What happens on `nodetool compact`? It's not keyspace specific.

Reply:
> What does "table_major_keyspace_compaction_task_impl" mean?

The task hierarchy is keyspace (top-level), shard, then table.

> What happens on `nodetool compact`? It's not keyspace specific.

Current JMX looks up all keyspaces and calls the api for each keyspace.
We could changes that by presenting a wildcard api to compact all keyspaces (we can add a flag for user only if it's interesting) and then do the uber-flush from there as an optimization.

Reply:
> > What does "table_major_keyspace_compaction_task_impl" mean?
> 
> The task hierarchy is keyspace (top-level), shard, then table.

How did you decode that from table_major_keyspace_compaction_task_impl?


"keyspace" is between "major" and "compaction".

> 
> > What happens on `nodetool compact`? It's not keyspace specific.
> 
> Currently, JMX looks up all keyspaces and calls the api for each keyspace. We could change that by presenting a wildcard api to compact all keyspaces (we can add a flag for user only if it's interesting) and then do the uber-flush from there as an optimization.

We might want to, with many keyspaces it would generate many flushes.

Reply:
I raised this concern above. Maybe we could restrict this system-wide flush before major to once a day. We also don't have control over how the user issues major (similar to repair jobs). There could be an application that calls nodetool compact for every table / keyspace at a time.

Reply:
> I raised this concern above. Maybe we could restrict this system-wide flush before major to once a day. We also don't have control over how the user issues major (similar to repair jobs). There could be an application that calls nodetool compact for every table / keyspace at a time.

There's a limit to the level of protection the the system needs to provide against anti-patterns.
Given a compact_all_keyspaces api, it can print a warning if it's called too frequently.  That's enough IMO.

Reply:
> There could be an application that calls nodetool compact for every table / keyspace at a time.

We can add a flag to disable all-tables flush, similar to the `skip flush` flag to the snapshot api.
In this case the user is taking responsibility to call (or not to call) `nodetool flush`.